{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Query:\n",
    "What genes encode proteins that physically interact with proteins encoded by the EDS associated genes.\n",
    "\n",
    "### Input:\n",
    "\n",
    "Hardcoded tsv file from:\n",
    "\n",
    "https://raw.githubusercontent.com/NCATS-Tangerine/cq-notebooks/master/FA_gene_sets/FA_1_core_complex.txt\n",
    "\n",
    "### Goal:\n",
    "This simple query aims to expand the FA-core gene set based PPI network membership.\n",
    "\n",
    "### Route:\n",
    "1. Protein-protein interactions from Monarch Solr index, either direct or inferred through orthology.  Sources: [BioGRID](https://thebiogrid.org/) and [STRING](http://string-db.org/)\n",
    "2. Gene-ortholog associations from Monarch Solr index.  Sources: [Panther](http://www.pantherdb.org/)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/putmanti/.virtualenvs/sandbox/lib/python3.9/site-packages/pandas/util/_decorators.py:311: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.\n",
      "  return func(*args, **kwargs)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NCBIGene:1289</td>\n",
       "      <td>COL5A1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NCBIGene:1303</td>\n",
       "      <td>COL12A1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NCBIGene:7148</td>\n",
       "      <td>TNXB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NCBIGene:4638</td>\n",
       "      <td>MYLK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NCBIGene:4629</td>\n",
       "      <td>MYH11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NCBIGene:1277</td>\n",
       "      <td>COL1A1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NCBIGene:1278</td>\n",
       "      <td>COL1A2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            gene   symbol\n",
       "0  NCBIGene:1289   COL5A1\n",
       "1  NCBIGene:1303  COL12A1\n",
       "2  NCBIGene:7148     TNXB\n",
       "3  NCBIGene:4638     MYLK\n",
       "4  NCBIGene:4629    MYH11\n",
       "5  NCBIGene:1277   COL1A1\n",
       "6  NCBIGene:1278   COL1A2"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/putmanti/.virtualenvs/sandbox/lib/python3.9/site-packages/pandas/util/_decorators.py:311: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.\n",
      "  return func(*args, **kwargs)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>interactor_id</th>\n",
       "      <th>interactor_symbol</th>\n",
       "      <th>qualifier</th>\n",
       "      <th>inferred_gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>COL5A1</td>\n",
       "      <td>HGNC:9255</td>\n",
       "      <td>PPIB</td>\n",
       "      <td>direct</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>COL5A1</td>\n",
       "      <td>HGNC:9081</td>\n",
       "      <td>PLOD1</td>\n",
       "      <td>direct</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>COL5A1</td>\n",
       "      <td>HGNC:9083</td>\n",
       "      <td>PLOD3</td>\n",
       "      <td>direct</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>COL5A1</td>\n",
       "      <td>HGNC:9082</td>\n",
       "      <td>PLOD2</td>\n",
       "      <td>direct</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>COL5A1</td>\n",
       "      <td>HGNC:16171</td>\n",
       "      <td>CHMP4B</td>\n",
       "      <td>direct</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>695</th>\n",
       "      <td>COL1A2</td>\n",
       "      <td>HGNC:6156</td>\n",
       "      <td>ITGB3</td>\n",
       "      <td>direct</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>696</th>\n",
       "      <td>COL1A2</td>\n",
       "      <td>HGNC:6153</td>\n",
       "      <td>ITGB1</td>\n",
       "      <td>direct</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>697</th>\n",
       "      <td>COL1A2</td>\n",
       "      <td>HGNC:6150</td>\n",
       "      <td>ITGAV</td>\n",
       "      <td>direct</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>698</th>\n",
       "      <td>COL1A2</td>\n",
       "      <td>HGNC:6134</td>\n",
       "      <td>ITGA1</td>\n",
       "      <td>direct</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>699</th>\n",
       "      <td>COL1A2</td>\n",
       "      <td>HGNC:6160</td>\n",
       "      <td>ITGB5</td>\n",
       "      <td>direct</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>700 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       gene interactor_id interactor_symbol qualifier inferred_gene\n",
       "0    COL5A1     HGNC:9255              PPIB    direct           NaN\n",
       "1    COL5A1     HGNC:9081             PLOD1    direct           NaN\n",
       "2    COL5A1     HGNC:9083             PLOD3    direct           NaN\n",
       "3    COL5A1     HGNC:9082             PLOD2    direct           NaN\n",
       "4    COL5A1    HGNC:16171            CHMP4B    direct           NaN\n",
       "..      ...           ...               ...       ...           ...\n",
       "695  COL1A2     HGNC:6156             ITGB3    direct           NaN\n",
       "696  COL1A2     HGNC:6153             ITGB1    direct           NaN\n",
       "697  COL1A2     HGNC:6150             ITGAV    direct           NaN\n",
       "698  COL1A2     HGNC:6134             ITGA1    direct           NaN\n",
       "699  COL1A2     HGNC:6160             ITGB5    direct           NaN\n",
       "\n",
       "[700 rows x 5 columns]"
      ]
     },
     "execution_count": 135,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import requests\n",
    "import pandas as pd\n",
    "import copy\n",
    "core_set = 'https://raw.githubusercontent.com/kcortes133/EDS_Rotation/master/genes.txt'\n",
    "\n",
    "columns = ['gene', 'interactor_id', 'interactor_symbol', 'qualifier', 'inferred_gene']\n",
    "dataframe = pd.read_csv(core_set, sep=\"   \", names=['gene', 'symbol']) # sep on 3 spaces\n",
    "dataframe\n",
    "solr_url = 'https://solr.monarchinitiative.org/solr/golr/select'\n",
    "def get_solr_results(solr, params):\n",
    "    resultCount = params['rows']\n",
    "    while params['start'] < resultCount:\n",
    "        solr_request = requests.get(solr, params=params)\n",
    "        response = solr_request.json()\n",
    "        resultCount = response['response']['numFound']\n",
    "        params['start'] += params['rows']\n",
    "        for doc in response['response']['docs']:\n",
    "            yield doc\n",
    "\n",
    "interaction_params = {\n",
    "    'wt': 'json',\n",
    "    'rows': 100,\n",
    "    'start': 0,\n",
    "    'q': '*:*',\n",
    "    'fl': 'subject, subject_label, subject_closure, \\\n",
    "           object, object_label, object_taxon',\n",
    "    'fq': ['relation_closure: \"RO:0002434\"']\n",
    "}\n",
    "\n",
    "# Make new dataframe for results\n",
    "interact_table = pd.DataFrame(columns=columns)\n",
    "\n",
    "\n",
    "# Get interactions, both direct and inferred\n",
    "for index, row in dataframe.iterrows():\n",
    "    params = copy.deepcopy(interaction_params)\n",
    "    params['fq'].append('subject_closure: \"{0}\" \\\n",
    "                        OR subject_ortholog_closure: \"{0}\"'\n",
    "                        .format(row['gene']))\n",
    "    for doc in get_solr_results(solr_url, params):\n",
    "        result = {}\n",
    "        result['gene'] = row['symbol']\n",
    "        result['interactor_id'] = doc['object']\n",
    "        result['interactor_symbol'] = doc['object_label']\n",
    "        if row['gene'] in doc['subject_closure']:\n",
    "            result['qualifier'] = \"direct\"\n",
    "        else:\n",
    "            result['qualifier'] = \"homology\"    \n",
    "        interact_table = interact_table.append(result, ignore_index=True)\n",
    "            \n",
    "interact_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [],
   "source": [
    "interact_table2 = interact_table[interact_table['gene'].isin([\"MYLK\", \"COL5A1\"])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [],
   "source": [
    "interactors = interact_table2[['gene', 'interactor_symbol']].drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_interactors = interactors.value_counts('interactor_symbol').reset_index()\n",
    "multi_interactors = all_interactors[all_interactors[0] > 1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>interactor_symbol</th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ITGB1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>VIRMA</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ANLN</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ITGA1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  interactor_symbol  0\n",
       "0             ITGB1  2\n",
       "1             VIRMA  2\n",
       "2              ANLN  2\n",
       "3             ITGA1  2"
      ]
     },
     "execution_count": 139,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "multi_interactors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [],
   "source": [
    "include_interactors = multi_interactors[\"interactor_symbol\"].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>interactor_id</th>\n",
       "      <th>interactor_symbol</th>\n",
       "      <th>qualifier</th>\n",
       "      <th>inferred_gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>545</th>\n",
       "      <td>COL1A1</td>\n",
       "      <td>HGNC:6664</td>\n",
       "      <td>LOX</td>\n",
       "      <td>direct</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>684</th>\n",
       "      <td>COL1A2</td>\n",
       "      <td>HGNC:6664</td>\n",
       "      <td>LOX</td>\n",
       "      <td>direct</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       gene interactor_id interactor_symbol qualifier inferred_gene\n",
       "545  COL1A1     HGNC:6664               LOX    direct           NaN\n",
       "684  COL1A2     HGNC:6664               LOX    direct           NaN"
      ]
     },
     "execution_count": 141,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "interact_table[interact_table[\"interactor_symbol\"] == \"LOX\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [],
   "source": [
    "import graphviz as gv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [],
   "source": [
    "g = gv.Digraph()\n",
    "nodes = list(set(interact_table['gene'].tolist())) + include_interactors\n",
    "for node in nodes:\n",
    "    g.node(node)\n",
    "for row in interact_table.itertuples():\n",
    "    if row.interactor_symbol in include_interactors:\n",
    "        g.edge(row.gene, row.interactor_symbol, label=\"interacts_with\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       " \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<!-- Generated by graphviz version 2.49.1 (20210923.0004)\n",
       " -->\n",
       "<!-- Pages: 1 -->\n",
       "<svg width=\"1660pt\" height=\"131pt\"\n",
       " viewBox=\"0.00 0.00 1660.40 131.00\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n",
       "<g id=\"graph0\" class=\"graph\" transform=\"scale(1 1) rotate(0) translate(4 127)\">\n",
       "<polygon fill=\"white\" stroke=\"transparent\" points=\"-4,4 -4,-127 1656.4,-127 1656.4,4 -4,4\"/>\n",
       "<!-- COL12A1 -->\n",
       "<g id=\"node1\" class=\"node\">\n",
       "<title>COL12A1</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" cx=\"49.4\" cy=\"-105\" rx=\"49.29\" ry=\"18\"/>\n",
       "<text text-anchor=\"middle\" x=\"49.4\" y=\"-101.3\" font-family=\"Times,serif\" font-size=\"14.00\">COL12A1</text>\n",
       "</g>\n",
       "<!-- VIRMA -->\n",
       "<g id=\"node9\" class=\"node\">\n",
       "<title>VIRMA</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" cx=\"152.4\" cy=\"-18\" rx=\"40.89\" ry=\"18\"/>\n",
       "<text text-anchor=\"middle\" x=\"152.4\" y=\"-14.3\" font-family=\"Times,serif\" font-size=\"14.00\">VIRMA</text>\n",
       "</g>\n",
       "<!-- COL12A1&#45;&gt;VIRMA -->\n",
       "<g id=\"edge5\" class=\"edge\">\n",
       "<title>COL12A1&#45;&gt;VIRMA</title>\n",
       "<path fill=\"none\" stroke=\"black\" d=\"M45.12,-86.69C43.59,-76.26 43.64,-63.26 50.4,-54 62.47,-37.45 82.86,-28.65 102.24,-24\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"103.16,-27.39 112.24,-21.93 101.74,-20.53 103.16,-27.39\"/>\n",
       "<text text-anchor=\"middle\" x=\"90.4\" y=\"-57.8\" font-family=\"Times,serif\" font-size=\"14.00\">interacts_with</text>\n",
       "</g>\n",
       "<!-- MYLK -->\n",
       "<g id=\"node2\" class=\"node\">\n",
       "<title>MYLK</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" cx=\"564.4\" cy=\"-105\" rx=\"37.89\" ry=\"18\"/>\n",
       "<text text-anchor=\"middle\" x=\"564.4\" y=\"-101.3\" font-family=\"Times,serif\" font-size=\"14.00\">MYLK</text>\n",
       "</g>\n",
       "<!-- ITGB1 -->\n",
       "<g id=\"node8\" class=\"node\">\n",
       "<title>ITGB1</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" cx=\"1285.4\" cy=\"-18\" rx=\"36.29\" ry=\"18\"/>\n",
       "<text text-anchor=\"middle\" x=\"1285.4\" y=\"-14.3\" font-family=\"Times,serif\" font-size=\"14.00\">ITGB1</text>\n",
       "</g>\n",
       "<!-- MYLK&#45;&gt;ITGB1 -->\n",
       "<g id=\"edge9\" class=\"edge\">\n",
       "<title>MYLK&#45;&gt;ITGB1</title>\n",
       "<path fill=\"none\" stroke=\"black\" d=\"M601.01,-100.6C642.36,-96.84 711.66,-90.83 771.4,-87 814.58,-84.23 1119.86,-84.13 1160.4,-69 1170.48,-65.24 1169.92,-59.1 1179.4,-54 1199.4,-43.24 1223.3,-34.99 1243.34,-29.22\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"1244.43,-32.55 1253.13,-26.5 1242.56,-25.8 1244.43,-32.55\"/>\n",
       "<text text-anchor=\"middle\" x=\"1219.4\" y=\"-57.8\" font-family=\"Times,serif\" font-size=\"14.00\">interacts_with</text>\n",
       "</g>\n",
       "<!-- MYLK&#45;&gt;VIRMA -->\n",
       "<g id=\"edge7\" class=\"edge\">\n",
       "<title>MYLK&#45;&gt;VIRMA</title>\n",
       "<path fill=\"none\" stroke=\"black\" d=\"M528.34,-99.63C489.55,-94.41 426.32,-84.4 373.4,-69 356.36,-64.04 353.42,-59.02 336.4,-54 291.35,-40.73 238.57,-31.29 201.12,-25.59\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"201.38,-22.09 190.98,-24.08 200.35,-29.01 201.38,-22.09\"/>\n",
       "<text text-anchor=\"middle\" x=\"413.4\" y=\"-57.8\" font-family=\"Times,serif\" font-size=\"14.00\">interacts_with</text>\n",
       "</g>\n",
       "<!-- ANLN -->\n",
       "<g id=\"node10\" class=\"node\">\n",
       "<title>ANLN</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" cx=\"461.4\" cy=\"-18\" rx=\"36\" ry=\"18\"/>\n",
       "<text text-anchor=\"middle\" x=\"461.4\" y=\"-14.3\" font-family=\"Times,serif\" font-size=\"14.00\">ANLN</text>\n",
       "</g>\n",
       "<!-- MYLK&#45;&gt;ANLN -->\n",
       "<g id=\"edge8\" class=\"edge\">\n",
       "<title>MYLK&#45;&gt;ANLN</title>\n",
       "<path fill=\"none\" stroke=\"black\" d=\"M527.3,-101.58C504.64,-97.97 477.23,-89.35 462.4,-69 457.7,-62.56 456.24,-54.31 456.31,-46.38\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"459.8,-46.56 457.12,-36.31 452.83,-46 459.8,-46.56\"/>\n",
       "<text text-anchor=\"middle\" x=\"502.4\" y=\"-57.8\" font-family=\"Times,serif\" font-size=\"14.00\">interacts_with</text>\n",
       "</g>\n",
       "<!-- ITGA1 -->\n",
       "<g id=\"node11\" class=\"node\">\n",
       "<title>ITGA1</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" cx=\"873.4\" cy=\"-18\" rx=\"37.09\" ry=\"18\"/>\n",
       "<text text-anchor=\"middle\" x=\"873.4\" y=\"-14.3\" font-family=\"Times,serif\" font-size=\"14.00\">ITGA1</text>\n",
       "</g>\n",
       "<!-- MYLK&#45;&gt;ITGA1 -->\n",
       "<g id=\"edge10\" class=\"edge\">\n",
       "<title>MYLK&#45;&gt;ITGA1</title>\n",
       "<path fill=\"none\" stroke=\"black\" d=\"M593.07,-93C608.82,-86.62 628.53,-78.06 645.4,-69 656.15,-63.23 657.03,-58.45 668.4,-54 720.26,-33.68 784.06,-25.14 826.58,-21.56\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"826.95,-25.04 836.65,-20.77 826.4,-18.06 826.95,-25.04\"/>\n",
       "<text text-anchor=\"middle\" x=\"708.4\" y=\"-57.8\" font-family=\"Times,serif\" font-size=\"14.00\">interacts_with</text>\n",
       "</g>\n",
       "<!-- MYH11 -->\n",
       "<g id=\"node3\" class=\"node\">\n",
       "<title>MYH11</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" cx=\"821.4\" cy=\"-105\" rx=\"40.89\" ry=\"18\"/>\n",
       "<text text-anchor=\"middle\" x=\"821.4\" y=\"-101.3\" font-family=\"Times,serif\" font-size=\"14.00\">MYH11</text>\n",
       "</g>\n",
       "<!-- MYH11&#45;&gt;ANLN -->\n",
       "<g id=\"edge11\" class=\"edge\">\n",
       "<title>MYH11&#45;&gt;ANLN</title>\n",
       "<path fill=\"none\" stroke=\"black\" d=\"M798.75,-89.99C788.97,-83.76 777.48,-76.22 767.4,-69 758.65,-62.73 758.33,-58.14 748.4,-54 705.87,-36.28 576.75,-25.97 507.1,-21.56\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"507.06,-18.05 496.86,-20.93 506.63,-25.04 507.06,-18.05\"/>\n",
       "<text text-anchor=\"middle\" x=\"807.4\" y=\"-57.8\" font-family=\"Times,serif\" font-size=\"14.00\">interacts_with</text>\n",
       "</g>\n",
       "<!-- MYH11&#45;&gt;ITGA1 -->\n",
       "<g id=\"edge12\" class=\"edge\">\n",
       "<title>MYH11&#45;&gt;ITGA1</title>\n",
       "<path fill=\"none\" stroke=\"black\" d=\"M836.8,-88.05C841.84,-82.32 847.22,-75.64 851.4,-69 856.01,-61.66 860.14,-53.19 863.53,-45.33\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"866.87,-46.41 867.42,-35.83 860.39,-43.76 866.87,-46.41\"/>\n",
       "<text text-anchor=\"middle\" x=\"899.4\" y=\"-57.8\" font-family=\"Times,serif\" font-size=\"14.00\">interacts_with</text>\n",
       "</g>\n",
       "<!-- COL1A1 -->\n",
       "<g id=\"node4\" class=\"node\">\n",
       "<title>COL1A1</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" cx=\"1128.4\" cy=\"-105\" rx=\"44.69\" ry=\"18\"/>\n",
       "<text text-anchor=\"middle\" x=\"1128.4\" y=\"-101.3\" font-family=\"Times,serif\" font-size=\"14.00\">COL1A1</text>\n",
       "</g>\n",
       "<!-- COL1A1&#45;&gt;ITGB1 -->\n",
       "<g id=\"edge13\" class=\"edge\">\n",
       "<title>COL1A1&#45;&gt;ITGB1</title>\n",
       "<path fill=\"none\" stroke=\"black\" d=\"M1172.73,-102.24C1201.51,-98.95 1238.22,-90.42 1263.4,-69 1270.46,-62.99 1275.33,-54.28 1278.66,-45.81\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"1282.04,-46.74 1281.9,-36.15 1275.4,-44.52 1282.04,-46.74\"/>\n",
       "<text text-anchor=\"middle\" x=\"1313.4\" y=\"-57.8\" font-family=\"Times,serif\" font-size=\"14.00\">interacts_with</text>\n",
       "</g>\n",
       "<!-- COL1A1&#45;&gt;ITGA1 -->\n",
       "<g id=\"edge14\" class=\"edge\">\n",
       "<title>COL1A1&#45;&gt;ITGA1</title>\n",
       "<path fill=\"none\" stroke=\"black\" d=\"M1087.97,-97.14C1057.16,-91.32 1013.95,-81.85 977.4,-69 970.22,-66.48 934.97,-49.36 907.48,-35.85\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"909.02,-32.7 898.5,-31.43 905.93,-38.98 909.02,-32.7\"/>\n",
       "<text text-anchor=\"middle\" x=\"1017.4\" y=\"-57.8\" font-family=\"Times,serif\" font-size=\"14.00\">interacts_with</text>\n",
       "</g>\n",
       "<!-- TNXB -->\n",
       "<g id=\"node5\" class=\"node\">\n",
       "<title>TNXB</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" cx=\"1594.4\" cy=\"-105\" rx=\"36\" ry=\"18\"/>\n",
       "<text text-anchor=\"middle\" x=\"1594.4\" y=\"-101.3\" font-family=\"Times,serif\" font-size=\"14.00\">TNXB</text>\n",
       "</g>\n",
       "<!-- TNXB&#45;&gt;ITGB1 -->\n",
       "<g id=\"edge6\" class=\"edge\">\n",
       "<title>TNXB&#45;&gt;ITGB1</title>\n",
       "<path fill=\"none\" stroke=\"black\" d=\"M1586.2,-87.22C1579.73,-75.88 1569.61,-61.65 1556.4,-54 1519.23,-32.49 1399.07,-23.9 1331.91,-20.71\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"1331.75,-17.2 1321.61,-20.24 1331.44,-24.19 1331.75,-17.2\"/>\n",
       "<text text-anchor=\"middle\" x=\"1612.4\" y=\"-57.8\" font-family=\"Times,serif\" font-size=\"14.00\">interacts_with</text>\n",
       "</g>\n",
       "<!-- COL1A2 -->\n",
       "<g id=\"node6\" class=\"node\">\n",
       "<title>COL1A2</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" cx=\"1439.4\" cy=\"-105\" rx=\"44.69\" ry=\"18\"/>\n",
       "<text text-anchor=\"middle\" x=\"1439.4\" y=\"-101.3\" font-family=\"Times,serif\" font-size=\"14.00\">COL1A2</text>\n",
       "</g>\n",
       "<!-- COL1A2&#45;&gt;ITGB1 -->\n",
       "<g id=\"edge15\" class=\"edge\">\n",
       "<title>COL1A2&#45;&gt;ITGB1</title>\n",
       "<path fill=\"none\" stroke=\"black\" d=\"M1458.78,-88.33C1469.16,-78.05 1478.17,-64.61 1469.4,-54 1452.22,-33.24 1380.18,-24.69 1331.5,-21.24\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"1331.71,-17.75 1321.5,-20.58 1331.25,-24.74 1331.71,-17.75\"/>\n",
       "<text text-anchor=\"middle\" x=\"1512.4\" y=\"-57.8\" font-family=\"Times,serif\" font-size=\"14.00\">interacts_with</text>\n",
       "</g>\n",
       "<!-- COL1A2&#45;&gt;ITGA1 -->\n",
       "<g id=\"edge16\" class=\"edge\">\n",
       "<title>COL1A2&#45;&gt;ITGA1</title>\n",
       "<path fill=\"none\" stroke=\"black\" d=\"M1416.54,-89.3C1394.57,-75.3 1363.79,-56.05 1357.4,-54 1316.5,-40.91 1033.75,-26.47 920.32,-21.13\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"920.36,-17.63 910.2,-20.66 920.03,-24.62 920.36,-17.63\"/>\n",
       "<text text-anchor=\"middle\" x=\"1423.4\" y=\"-57.8\" font-family=\"Times,serif\" font-size=\"14.00\">interacts_with</text>\n",
       "</g>\n",
       "<!-- COL5A1 -->\n",
       "<g id=\"node7\" class=\"node\">\n",
       "<title>COL5A1</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" cx=\"359.4\" cy=\"-105\" rx=\"44.69\" ry=\"18\"/>\n",
       "<text text-anchor=\"middle\" x=\"359.4\" y=\"-101.3\" font-family=\"Times,serif\" font-size=\"14.00\">COL5A1</text>\n",
       "</g>\n",
       "<!-- COL5A1&#45;&gt;ITGB1 -->\n",
       "<g id=\"edge3\" class=\"edge\">\n",
       "<title>COL5A1&#45;&gt;ITGB1</title>\n",
       "<path fill=\"none\" stroke=\"black\" d=\"M401.57,-98.61C433.35,-94.75 478.04,-89.77 517.4,-87 577.28,-82.79 1001.02,-89.63 1057.4,-69 1067.5,-65.3 1066.56,-58.36 1076.4,-54 1129.02,-30.68 1195.39,-22.79 1239.04,-20.18\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"1239.28,-23.68 1249.08,-19.65 1238.91,-16.69 1239.28,-23.68\"/>\n",
       "<text text-anchor=\"middle\" x=\"1116.4\" y=\"-57.8\" font-family=\"Times,serif\" font-size=\"14.00\">interacts_with</text>\n",
       "</g>\n",
       "<!-- COL5A1&#45;&gt;VIRMA -->\n",
       "<g id=\"edge1\" class=\"edge\">\n",
       "<title>COL5A1&#45;&gt;VIRMA</title>\n",
       "<path fill=\"none\" stroke=\"black\" d=\"M315.34,-101.23C265.57,-97.23 188.73,-88.09 167.4,-69 160.86,-63.16 157.13,-54.6 155.01,-46.2\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"158.42,-45.43 153.14,-36.24 151.54,-46.72 158.42,-45.43\"/>\n",
       "<text text-anchor=\"middle\" x=\"207.4\" y=\"-57.8\" font-family=\"Times,serif\" font-size=\"14.00\">interacts_with</text>\n",
       "</g>\n",
       "<!-- COL5A1&#45;&gt;ANLN -->\n",
       "<g id=\"edge2\" class=\"edge\">\n",
       "<title>COL5A1&#45;&gt;ANLN</title>\n",
       "<path fill=\"none\" stroke=\"black\" d=\"M317.46,-98.31C280.68,-91.59 236.01,-78.16 256.4,-54 276.36,-30.34 361.53,-22.68 415.61,-20.19\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"415.81,-23.69 425.66,-19.77 415.52,-16.69 415.81,-23.69\"/>\n",
       "<text text-anchor=\"middle\" x=\"296.4\" y=\"-57.8\" font-family=\"Times,serif\" font-size=\"14.00\">interacts_with</text>\n",
       "</g>\n",
       "<!-- COL5A1&#45;&gt;ITGA1 -->\n",
       "<g id=\"edge4\" class=\"edge\">\n",
       "<title>COL5A1&#45;&gt;ITGA1</title>\n",
       "<path fill=\"none\" stroke=\"black\" d=\"M403.7,-101.45C441.92,-97.89 497.82,-89.37 542.4,-69 552.18,-64.53 551.44,-58.09 561.4,-54 608.22,-34.75 751.37,-25.01 826.4,-21.11\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"826.71,-24.6 836.52,-20.6 826.35,-17.61 826.71,-24.6\"/>\n",
       "<text text-anchor=\"middle\" x=\"601.4\" y=\"-57.8\" font-family=\"Times,serif\" font-size=\"14.00\">interacts_with</text>\n",
       "</g>\n",
       "</g>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<graphviz.dot.Digraph at 0x1278f4d90>"
      ]
     },
     "execution_count": 144,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NCBIGene:1289</td>\n",
       "      <td>COL5A1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NCBIGene:1303</td>\n",
       "      <td>COL12A1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NCBIGene:7148</td>\n",
       "      <td>TNXB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NCBIGene:4638</td>\n",
       "      <td>MYLK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NCBIGene:4629</td>\n",
       "      <td>MYH11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NCBIGene:1277</td>\n",
       "      <td>COL1A1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NCBIGene:1278</td>\n",
       "      <td>COL1A2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            gene   symbol\n",
       "0  NCBIGene:1289   COL5A1\n",
       "1  NCBIGene:1303  COL12A1\n",
       "2  NCBIGene:7148     TNXB\n",
       "3  NCBIGene:4638     MYLK\n",
       "4  NCBIGene:4629    MYH11\n",
       "5  NCBIGene:1277   COL1A1\n",
       "6  NCBIGene:1278   COL1A2"
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (1920336974.py, line 2)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  File \u001b[0;32m\"/var/folders/lm/x4jknky50yx5flf2vznggl380000gp/T/ipykernel_39161/1920336974.py\"\u001b[0;36m, line \u001b[0;32m2\u001b[0m\n\u001b[0;31m    return graph of eds genes in list and only interactors that they share\u001b[0m\n\u001b[0m                 ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "def pairwise_graph(list_of_genes):\n",
    "    return graph of eds genes in list and only interactors that they share\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
